app/commands/v2/unpublish.rb

```diff
@@ -149,7 +150,21 @@ def find_unpublishable_content_item(content_id)
         end
 
         filter = ContentItemFilter.new(scope: ContentItem.where(content_id: content_id))
-        filter.filter(locale: locale, state: allowed_states).first
+        filter.filter(locale: locale, state: allowed_states).last
+      end
+
+      def supersede_previous_items(content_item)
+        items = ContentItemFilter.similar_to(
+          content_item,
+          state: %w(published unpublished),
+          base_path: nil,
+          user_version: nil,
+        )
+
+        items.each do |item|
+          next if item == content_item
+          State.supersede(item)
```

Comment:
On 11/07/16 11:48, Elliot Crosby-McCullough wrote:
>> @@ -149,7 +150,21 @@ def find_unpublishable_content_item(content_id)
>>          end
>>  
>>          filter = ContentItemFilter.new(scope: ContentItem.where(content_id: content_id))
>> -        filter.filter(locale: locale, state: allowed_states).first
>> +        filter.filter(locale: locale, state: allowed_states).last
>> +      end
>> +
>> +      def supersede_previous_items(content_item)
>> +        items = ContentItemFilter.similar_to(
>> +          content_item,
>> +          state: %w(published unpublished),
>> +          base_path: nil,
>> +          user_version: nil,
>> +        )
>> +
>> +        items.each do |item|
>> +          next if item == content_item
>> +          State.supersede(item)
> 
> We don't have any validations which ensure this is the case (yet, there's a ticket in the backlog) so for the sake of robustness we agreed to take the more careful approach.

I would say that the most careful approach would be to check that
reality matches the assumptions (only one item to supersede), and error
if it does not.

As I see it, the lack of confidence that this assumption holds makes it
more important to do this, not less.